tests/test_rtdc_config.py

```diff
@@ -81,6 +84,267 @@ def test_config_update():
+                                 "channel information": "information"}
+
+
+def test_user_section_set_save_reload_fmt_tdms():
+    """Check that 'user' section metadata works for RTDC_TDMS"""
+    h5path = retrieve_data("rtdc_data_traces_video.zip")
+    metadata = {"channel area": 100.5,
+                "inlet": True,
+                "n_constrictions": 3,
+                "channel information": "other information"}
+    with new_dataset(h5path) as ds:
+        ds.config.update({"user": metadata})
+        expath = h5path.with_name("exported.rtdc")
+        ds.export.hdf5(expath, features=ds.features_innate)
+    # make sure that worked
+    with h5py.File(expath, "r") as h5:
+        assert h5.attrs["user:channel area"] == 100.5
+        assert h5.attrs["user:inlet"]
+        assert h5.attrs["user:n_constrictions"] == 3
+        assert h5.attrs["user:channel information"] == "other information"
+    # now check again with dclab
+    with new_dataset(expath) as ds2:
+        assert ds2.config["user"] == metadata
+
+
+def test_user_section_set_save_reload_fmt_hierarchy():
+    """Save the user section config information and reload it"""
+    h5path = retrieve_data("rtdc_data_hdf5_rtfdc.zip")
+    metadata = {"channel area": 100.5,
+                "inlet": True,
+                "n_constrictions": 3,
+                "channel information": "other information"}
+    with new_dataset(h5path) as ds:
+        ds.config.update({"user": metadata})
+        ch = new_dataset(ds)
+        expath = h5path.with_name("exported.rtdc")
+        ch.export.hdf5(expath, features=ch.features_innate)
+    # make sure that worked
+    with h5py.File(expath, "r") as h5:
+        assert h5.attrs["user:channel area"] == 100.5
+        assert h5.attrs["user:inlet"]
+        assert h5.attrs["user:n_constrictions"] == 3
+        assert h5.attrs["user:channel information"] == "other information"
+    # now check again with dclab
+    with new_dataset(expath) as ds2:
+        assert ds2.config["user"] == metadata
+
+
+# def test_user_section_set_save_reload_fmt_dict():
+#     """Check that 'user' section metadata works for RTDC_Dict"""
+#     ddict = example_data_dict(size=67, keys=["area_um", "deform"])
+#     ds = new_dataset(ddict)
+#     metadata = {"some metadata": 42}
+#     ds.config.update({"user": metadata})
+#     assert ds.config["user"] == metadata
+#     expath = "exported.rtdc"
+#     ds.export.hdf5(expath, features=ds.features_innate,
+#                    override=True)
+#     # make sure that worked
+#     with h5py.File(expath, "r") as h5:
+#         assert h5.attrs["user:some metadata"] == 42
+#     # fails due to RTDC_HDF5 trying to add self.title
+#     with pytest.raises(KeyError):
+#         with new_dataset(expath) as ds2:
+#             assert ds2.config["user"] == metadata
+#
+#
+# def test_user_section_set_save_reload_fmt_dcor():
+#     """Check that 'user' section metadata works for RTDC_Dcor"""
```

Comment:
This test hangs so need to do more checks on it.

Reply:
Probably hangs because it downloads and exports *all* features. Just use e.g. `features=["area_um", "deform"]`.